dash_labs/plugins/pages.py

```diff
@@ -170,16 +170,16 @@ def register_page(
         # Override the layout found in the file set during `plug`
         dash.page_registry[module]["layout"] = layout
 
-    # Reset order
-    order = []
-    for page_module in dash.page_registry:
-        if page["supplied_path"] == "/":
-            order.append((0, page_module))
-            page["order"] = 0
+    # set home page order
+    order_supplied = any(p["supplied_order"] for p in dash.page_registry.values())
+    page["order"] = 0 if page["path"] == "/" and not order_supplied else page["order"]
```

Comment:
Unless `supplied_order` is provided or not provided consistently, I think the result of this is going to depend on the order pages are walked - so for example changing a filename could affect this without any other consequences.

Can we move this part into `plug` and run it after we've finished walking all the pages?

Reply:
Yes, I tried that - it seems like it would be more efficient too.  However, that doesn't work when `dash.register_page()` is called after the app is instantiated like in [this example](https://github.com/plotly/dash-labs/blob/main/docs/demos/multi-page-basics/app.py#L8).  Those pages are not sorted properly.

To avoid any conflict (like someone specifying `order=0` in one app and `path="/"` in a different app, the rule is “if any order is supplied, then it’s order + alphanumerical. But if no order was supplied, then / will be first”

Reply:
I see - so following exactly your proposed conflict scenario, with post-registration like the example you linked:
```py
dash.register_page("home", layout="We're home!", path="/")
dash.register_page("very_important", layout="Don't miss it!", path="/important", order=0)
```
The first call will see `order_supplied=False` (because `dash.page_registry` is empty) so it will get `order=0` filled in, then the second one will also get `order=0` as that was explicitly supplied, and `home` will wind up sorted before `very_important`. But if we swapped the order of those two calls, `home` would NOT get `order=0`, so `very_important` would come first. Am I understanding this correctly?

So perhaps we can get around this with a hybrid of what you had before and now:
```py
order_supplied = any(p["supplied_order"] for p in dash.page_registry.values())
for p in dash.page_registry.values():
    # Note: use only supplied_order as input, order is just an output
    p["order"] = 0 if p["path"] == "/" and not order_supplied else p["supplied_order"]
```

Reply:
Oh, nice catch - that's what I intended.

The other thing I just found... if the only supplied order is `order=0`, then this is false :slightly_frowning_face: 
```
order_supplied = any(p["supplied_order"] for p in dash.page_registry.values())
```
Writing it like this works but it's not as pretty.  Is there a better way? :lipstick: 

```
order_supplied = False
    for p in dash.page_registry.values():
        if p["supplied_order"] is not None and p["supplied_order"] >= 0:
            order_supplied = True
            break
```

Reply:
ah right - yes, you can include `is not None and >= 0` in the one-liner, though it starts to get a little long:
```py
order_supplied = any(p["supplied_order"] is not None and p["supplied_order"] >= 0 for p in dash.page_registry.values())
```

Do you actually want to exclude negative orders though? Shouldn't `p["supplied_order"] is not None` be sufficient to mean "you've supplied an order somewhere so don't fill in a default anywhere"?

Reply:
Yes,  it's better not to limit it to >= 0
OK, I did a commit for the changes.